File: cloudera_training/devsh/exercises/spark-streaming-kafka/streaminglogskafka_project/src/main/scala/stubs/StreamingLogsKafka.scala
package stubs

import org.apache.spark.SparkContext
import org.apache.spark.streaming.StreamingContext
import org.apache.spark.streaming.Seconds
import org.apache.spark.streaming.kafka._
import kafka.serializer.StringDecoder

object StreamingLogsKafka {
  def main(args: Array[String]) {
    if (args.length < 1) {
      System.err.println("Usage: stubs.StreamingLogsKafka <topic>")
      System.exit(1)
    }  
    val topic = args(0)

    val sc = new SparkContext()
    sc.setLogLevel("ERROR")
 
     // Configure the Streaming Context with a 1 second batch duration
    val ssc = new StreamingContext(sc,Seconds(1))
       
    // TODO
    println("Stub not yet implemented")

    ssc.start()
    ssc.awaitTermination()
        
  }
}

File: cloudera_training/devsh/exercises/spark-streaming-kafka/stubs-python/StreamingLogsKafka.py
import sys
from pyspark import SparkContext
from pyspark.streaming import StreamingContext
from pyspark.streaming.kafka import KafkaUtils

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print >> sys.stderr, "Usage: StreamingLogsKafka.py <topic>"
        sys.exit(-1)
    
    topic = sys.argv[1]
     
    sc = SparkContext()   
    sc.setLogLevel("ERROR")

    # Configure the Streaming Context with a 1 second batch duration
    ssc = StreamingContext(sc,1)
        
    # TODO
    print "Stub not yet implemented"

    ssc.start()
    ssc.awaitTermination()